app/components/memorable_date_component.html.erb

```diff
@@ -107,11 +110,12 @@
                   minLength: 4,
                   maxLength: 4,
                   aria: {
-                    invalid: false,
+                    invalid: has_errors?,
                     labelledby: [
                       "memorable-date-year-label-#{unique_id}",
                       "memorable-date-year-hint-#{unique_id}",
                     ],
+                    describedby: ["validated-field-error-#{unique_id}"],
```

Comment:
How does this interact with the screen reader relative to the current JavaScript validation, both when an error is present and absent?